update of mingw-w64-x86_64-make (4.4.1-3 -> 4.4.1-4) breaks --output-sync=target and --output-sync=line, when parallel building+recursion

### Description / Steps to reproduce the issue

Since I have updated from **mingw-w64-x86_64-make** **4.4.1-3** to **4.4.1-4**, using `--output-sync=target` or `--output-sync=line` results in an error and weird output.

How to reproduce:

1. Use a project with recursion. For example the attached: [test_9ee52ff4ffede5c352a57f772b2fd3797f9199e5.zip](https://github.com/user-attachments/files/27177226/test_9ee52ff4ffede5c352a57f772b2fd3797f9199e5.zip)
Note: it simply contains of two very minimal Makefiles:
   -  Makefile:
   ```
   all:
   	@$(MAKE) -C sub
   ```

    - sub/Makefile
   ```
   all:
   	@echo hi from sub
   ```
2. Call `/mingw64/bin/mingw32-make.exe -j2 --output-sync=target` or `/mingw64/bin/mingw32-make.exe -j2 --output-sync=line`
3. This results in
    - a) An error `mingw32-make: *** [Makefile:2: all] Error 2`
    - b) It prints the help. At least in this minimal failing example - I have also seen scenarios where no help was printed. For full printout see section "Actual behavior".

Notes:
- This does not happen when using `--output-sync=recurse` - that still seems to work.


### Expected behavior

This is how this looked like for version 3:
```
$ /mingw64/bin/mingw32-make -j2 --output-sync=target
mingw32-make[1]: Entering directory 'C:/Users/robing/Desktop/test/sub'
hi from sub
mingw32-make[1]: Leaving directory 'C:/Users/robing/Desktop/test/sub'
```
and
```
$ /mingw64/bin/mingw32-make -j2 --output-sync=line
mingw32-make[1]: Entering directory 'C:/Users/robing/Desktop/test/sub'
hi from sub
mingw32-make[1]: Leaving directory 'C:/Users/robing/Desktop/test/sub'
```

### Actual behavior

For version 4 we get:
```
$ /mingw64/bin/mingw32-make -j2 --output-sync=target
mingw32-make: the '--sync-mutex' option requires a non-empty string argument
Usage: mingw32-make [options] [target] ...
Options:
  -b, -m                      Ignored for compatibility.
  -B, --always-make           Unconditionally make all targets.
  -C DIRECTORY, --directory=DIRECTORY
                              Change to DIRECTORY before doing anything.
  -d                          Print lots of debugging information.
  --debug[=FLAGS]             Print various types of debugging information.
  -e, --environment-overrides
                              Environment variables override makefiles.
  -E STRING, --eval=STRING    Evaluate STRING as a makefile statement.
  -f FILE, --file=FILE, --makefile=FILE
                              Read FILE as a makefile.
  -h, --help                  Print this message and exit.
  -i, --ignore-errors         Ignore errors from recipes.
  -I DIRECTORY, --include-dir=DIRECTORY
                              Search DIRECTORY for included makefiles.
  -j [N], --jobs[=N]          Allow N jobs at once; infinite jobs with no arg.
  --jobserver-style=STYLE     Select the style of jobserver to use.
  -k, --keep-going            Keep going when some targets can't be made.
  -l [N], --load-average[=N], --max-load[=N]
                              Don't start multiple jobs unless load is below N.
  -L, --check-symlink-times   Use the latest mtime between symlinks and target.
  -n, --just-print, --dry-run, --recon
                              Don't actually run any recipe; just print them.
  -o FILE, --old-file=FILE, --assume-old=FILE
                              Consider FILE to be very old and don't remake it.
  -O[TYPE], --output-sync[=TYPE]
                              Synchronize output of parallel jobs by TYPE.
  -p, --print-data-base       Print make's internal database.
  -q, --question              Run no recipe; exit status says if up to date.
  -r, --no-builtin-rules      Disable the built-in implicit rules.
  -R, --no-builtin-variables  Disable the built-in variable settings.
  --shuffle[={SEED|random|reverse|none}]
                              Perform shuffle of prerequisites and goals.
  -s, --silent, --quiet       Don't echo recipes.
  --no-silent                 Echo recipes (disable --silent mode).
  -S, --no-keep-going, --stop
                              Turns off -k.
  -t, --touch                 Touch targets instead of remaking them.
  --trace                     Print tracing information.
  -v, --version               Print the version number of make and exit.
  -w, --print-directory       Print the current directory.
  --no-print-directory        Turn off -w, even if it was turned on implicitly.
  -W FILE, --what-if=FILE, --new-file=FILE, --assume-new=FILE
                              Consider FILE to be infinitely new.
  --warn-undefined-variables  Warn when an undefined variable is referenced.

This program built for x86_64-w64-mingw32
Report bugs to <bug-make@gnu.org>
mingw32-make: *** [Makefile:2: all] Error 2
```
and
```
$ /mingw64/bin/mingw32-make -j2 --output-sync=line
: option '--sync-mutex' requires an argument
Usage: mingw32-make [options] [target] ...
Options:
  -b, -m                      Ignored for compatibility.
  -B, --always-make           Unconditionally make all targets.
  -C DIRECTORY, --directory=DIRECTORY
                              Change to DIRECTORY before doing anything.
  -d                          Print lots of debugging information.
  --debug[=FLAGS]             Print various types of debugging information.
  -e, --environment-overrides
                              Environment variables override makefiles.
  -E STRING, --eval=STRING    Evaluate STRING as a makefile statement.
  -f FILE, --file=FILE, --makefile=FILE
                              Read FILE as a makefile.
  -h, --help                  Print this message and exit.
  -i, --ignore-errors         Ignore errors from recipes.
  -I DIRECTORY, --include-dir=DIRECTORY
                              Search DIRECTORY for included makefiles.
  -j [N], --jobs[=N]          Allow N jobs at once; infinite jobs with no arg.
  --jobserver-style=STYLE     Select the style of jobserver to use.
  -k, --keep-going            Keep going when some targets can't be made.
  -l [N], --load-average[=N], --max-load[=N]
                              Don't start multiple jobs unless load is below N.
  -L, --check-symlink-times   Use the latest mtime between symlinks and target.
  -n, --just-print, --dry-run, --recon
                              Don't actually run any recipe; just print them.
  -o FILE, --old-file=FILE, --assume-old=FILE
                              Consider FILE to be very old and don't remake it.
  -O[TYPE], --output-sync[=TYPE]
                              Synchronize output of parallel jobs by TYPE.
  -p, --print-data-base       Print make's internal database.
  -q, --question              Run no recipe; exit status says if up to date.
  -r, --no-builtin-rules      Disable the built-in implicit rules.
  -R, --no-builtin-variables  Disable the built-in variable settings.
  --shuffle[={SEED|random|reverse|none}]
                              Perform shuffle of prerequisites and goals.
  -s, --silent, --quiet       Don't echo recipes.
  --no-silent                 Echo recipes (disable --silent mode).
  -S, --no-keep-going, --stop
                              Turns off -k.
  -t, --touch                 Touch targets instead of remaking them.
  --trace                     Print tracing information.
  -v, --version               Print the version number of make and exit.
  -w, --print-directory       Print the current directory.
  --no-print-directory        Turn off -w, even if it was turned on implicitly.
  -W FILE, --what-if=FILE, --new-file=FILE, --assume-new=FILE
                              Consider FILE to be infinitely new.
  --warn-undefined-variables  Warn when an undefined variable is referenced.

This program built for x86_64-w64-mingw32
Report bugs to <bug-make@gnu.org>
mingw32-make: *** [Makefile:2: all] Error 2
```

### Verification

- [x] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

MSYS_NT-10.0-22631

### MINGW environments affected

- [x] MINGW64
- [ ] MINGW32
- [ ] UCRT64
- [ ] CLANG64
- [ ] CLANGARM64

### Are you willing to submit a PR?

_No response_

--- a/mingw-w64-make/007-fix-w32-printf-modifier.patch
+++ b/mingw-w64-make/007-fix-w32-printf-modifier.patch
@@ -1,0 +1,26 @@
+Fix MS-specific format specifier in osync_get_mutex.
+
+The %I length modifier alone is ambiguous; use %I64x to match the
+(unsigned long long) cast. Otherwise osync_get_mutex in w32os.c writes
+a string that is either empty or garbage, and the following error
+appears:
+    *** invalid output sync mutex: <garbage>.  Stop.
+
+or, when the string starts with NUL:
+
+    : option '--sync-mutex' requires an argument
+
+Upstream fix:
+https://git.savannah.gnu.org/cgit/make.git/commit/?id=b3802782de3eff2c0f1eda9e7c0befd8cd142162
+
+--- a/src/w32/w32os.c
++++ b/src/w32/w32os.c
+@@ -433,7 +433,7 @@ osync_get_mutex ()
+       /* Prepare the mutex handle string for our children.
+          2 hex digits per byte + 2 characters for "0x" + null.  */
+       mutex = xmalloc ((2 * sizeof (osync_handle)) + 2 + 1);
+-      sprintf (mutex, "0x%Ix", (unsigned long long)(DWORD_PTR)osync_handle);
++      sprintf (mutex, "0x%I64x", (unsigned long long)(DWORD_PTR)osync_handle);
+     }
+
+   return mutex;

--- a/mingw-w64-make/PKGBUILD
+++ b/mingw-w64-make/PKGBUILD
@@ -12,7 +12,7 @@ _prog_name=mingw32-make
 pkgbase=mingw-w64-${_realname}
 pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}")
 pkgver=4.4.1
-pkgrel=4
+pkgrel=5
 pkgdesc="GNU make utility to maintain groups of programs (mingw-w64)"
 arch=('any')
 mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clangarm64')
@@ -34,6 +34,7 @@ source=("https://ftp.gnu.org/gnu/${_realname}/${_realname}-${pkgver}.tar.gz"
         'make-4.2.1-Makefile.am-gcc-only-link-libgnumake-1.dll.a.patch'
         '005-fix-default-cxx.patch'
         '006-fix-build-with-gcc-15.patch'
+        '007-fix-w32-printf-modifier.patch'
         'make-check-load-feature.mak'
         'mk_temp.c')
 sha256sums=('dd16fb1d67bfab79a72f5e8390735c49e3e8e70b4945a15ab1f81ddb78658fb3'
@@ -43,6 +44,7 @@ sha256sums=('dd16fb1d67bfab79a72f5e8390735c49e3e8e70b4945a15ab1f81ddb78658fb3'
             'f28d1596ac8f60bbe6042f3467627646ae7e7243e4f73652f9682255758ecdb9'
             'bd6a597d76b35a48020dc335040c96f562439970595a21014469b73b3b6c805a'
             'c5125d642ca6ae9b5788648aa51ecc7a44e829cf443b44768c9a66c85d85689d'
+            '8dc48c3f1d8b02728157a6fba833accb251ba0bdbe7a401da4704700da4c486c'
             '3d6ece384425ff1c3f691efeb7c97e89bb11825bb6648995552dfcb52afe661d'
             '61d4f57c311cf2e27ccf4d1da847216730f304cf17c5c386721182ffe2d4b1aa')
 
@@ -71,7 +73,8 @@ prepare() {
     make-linebuf-mingw.patch \
     make-4.3_undef-HAVE_STRUCT_DIRENT_D_TYPE.patch \
     005-fix-default-cxx.patch \
-    006-fix-build-with-gcc-15.patch
+    006-fix-build-with-gcc-15.patch \
+    007-fix-w32-printf-modifier.patch
   # https://lists.gnu.org/archive/html/bug-make/2022-11/msg00017.html
   apply_patch_with_msg \
     make-4.4-timestamps.patch

